File: language/java_study/javaLearn/src/main/java/com/iamhotcenplotz/chapter14/exercises/Exercise03_.java
package com.iamhotcenplotz.chapter14.exercises;

import java.util.HashMap;

/**
 * @author Der Hotcenplotz
 * @version 1.0
 */
public class Exercise03_ {
    public static void main(String[] args) {
        HashMap<String, Integer> salary = new HashMap<>();

        salary.put("jack",650);
        salary.put("tom",1200);
        salary.put("jane",1500);
        salary.put("George",30000);

        System.out.println(salary);

        salary.put("jack",2600);
        System.out.println(salary);

        salary.replaceAll((k, v) -> salary.get(k) + 100);

        for (String key: salary.keySet()){
            System.out.println(key + ": " + salary.get(key));
        }
    }
}

File: language/scala_study/ScalaLearn/src/main/scala/com/hotcenplotz/chapter14/exercise_/Exercise03_.scala
package com.hotcenplotz.chapter14.exercise_

/**
 * @author: Der Hotcenplotz
 * @version： 1.0 
 * @date: 2025-05-26 21:28 
 * @description:
 * */
object Exercise03_ {
    def main(args: Array[String]): Unit = {
        println(sum(List(Option(1), Option(2), None)))
    }
    def sum(lst:List[Option[Int]]) = lst.map(_.getOrElse(0)).sum
}
